packages/client/src/workflow-client.ts

```diff
@@ -581,14 +582,19 @@ export class WorkflowClient extends BaseClient {
     }
   }
 
-  protected rethrowGrpcError(err: unknown, workflowExecution: WorkflowExecution, fallbackMessage: string): never {
+  protected rethrowGrpcError(err: unknown, fallbackMessage: string, workflowExecution?: WorkflowExecution): never {
     if (isServerErrorResponse(err)) {
       if (err.code === grpcStatus.NOT_FOUND) {
-        throw new WorkflowNotFoundError(
-          err.details ?? 'Workflow not found',
-          workflowExecution.workflowId,
-          workflowExecution.runId
-        );
+        const matcher = err.message.match(/^5 NOT_FOUND: Namespace (.*?) is not found./);
```

Comment:
Don’t parse the error message, use the gRPC response header as I suggested in the corresponding issue to get the structured error.